Update CoreNEURON documentation (#950)
neuronsimulator/nrn: docs/coreneuron/how-to/coreneuron.md

# Using CoreNEURON with NEURON

[CoreNEURON](https://github.com/BlueBrain/CoreNeuron) is a compute engine for the NEURON simulator optimised for both memory usage and computational speed on modern CPU/GPU architectures. The goals of CoreNEURON are:

* Simulating large network models
* Reduce memory usage
* Support for GPUs
* Optimisations like Vectorisation and memory layout (e.g. Structure-Of-Array)

CoreNEURON is designed as a library within the NEURON simulator and can transparently handle all spiking network simulations including gap junction coupling with the **fixed time step method**. In order to run a NEURON model with CoreNEURON:

> * MOD files should be [THREADSAFE](https://neuron.yale.edu/neuron/docs/multithread-parallelization)

Reviewer: It is my understanding THREADSAFE is a must, we *should* imply it :) 
```suggestion
* MOD files shall be [THREADSAFE](https://neuron.yale.edu/neuron/docs/multithread-parallelization)
```

> * If random number generator is used then Random123 should be used instead of MCellRan4

Reviewer: ```suggestion
* Random123 shall be used if a random generator is needed (instead of MCellRan4)
```
